### PR TITLE
Add documentation for newly added SELinux resources under Chef Infra Client

### PR DESCRIPTION
**Pull request by neha-p6**
Signed-off-by: Neha Pansare <neha.pansare@progress.com>

## Description

Add documentation for newly added 7 SELinux resources under Chef Infra Client.
**Kindly do not merge until Chef Infra Client version 18 release.**

## Related PRs
https://github.com/chef/chef/pull/12694

## Check List

- [x] Spell Check
- [x] Local build
- [x] Examine the local build
- [x] All tests pass

### CODE REVIEW

File: data/infra/resources/selinux_boolean.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- property: persistent
+  ruby_type: true, false
+  required: false
+  default_value: 'true'
+  description_list:
+  - markdown: Set to true for value setting to survive reboot.
+- property: value
+  ruby_type: Integer, String, true, false
+  required: true
+  allowed_values: '"off", "on"'
```
Comment (IanMadd) [inline data/infra/resources/selinux_boolean.yaml]: It looks like it accepts other values. The example below has `true`.

File: data/infra/resources/selinux_permissive.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  required: false
+  default_value: The resource block's name
+  description_list:
+  - markdown: The SELinux context to permit.
+examples: |
+  **Disable enforcement on Apache**:
+
+  ```ruby
+  selinux_permissive 'httpd_t' do
+    notifies :restart, 'service[httpd]'
```
Comment (IanMadd) [inline data/infra/resources/selinux_permissive.yaml:43]: Is this correct? I'm not seeing notifies or restart in the text above.

**Review by IanMadd — COMMENTED** [review]
See feedback @neha-p6

File: data/infra/resources/selinux_permissive.yaml (hunk 2 above)
Comment (neha-p6) [reply data/infra/resources/selinux_permissive.yaml:43]: Yeah, `notifies`  would be inherited from `Resource` as common functionality. The default action for this resource is `add` so example was to show disabling enforcement on this service and then firing notifies event. Let me know your thoughts

File: data/infra/resources/selinux_boolean.yaml (hunk 1 above)
Comment (neha-p6) [reply data/infra/resources/selinux_boolean.yaml]: Yeah, it accepts `Integer, String, true, false` and by `allowed_values` it means what ever is passed in either of those data formats, if valid, it is converted to either "on" or "off"
Better description you  could suggest?

File: data/infra/resources/selinux_boolean.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- property: persistent
+  ruby_type: true, false
+  required: false
+  default_value: 'true'
+  description_list:
+  - markdown: Set to `true` for value setting to survive reboot.
+- property: value
+  ruby_type: Integer, String, true, false
+  required: true
+  allowed_values: '"off", "on"'
```
Comment (IanMadd) [inline data/infra/resources/selinux_boolean.yaml]:
```suggestion
  allowed_values: '"on", "off", true, false, 0, 1'
```

File: data/infra/resources/selinux_permissive.yaml (hunk 2 above)
Comment (IanMadd) [reply data/infra/resources/selinux_permissive.yaml:43]: Ok, I see.

**Review by IanMadd — APPROVED** [review]

**Pull request merged**